File: examples/lua/lgi/get-basic-nm-info.lua
#!/usr/bin/env lua
-- SPDX-License-Identifier: GPL-2.0+
--
-- Copyright 2015 Red Hat, Inc.
--
--
-- This example gets basic information about NetworkManager.
-- The example uses libnm library using GObject introspection via Lua lgi module.
-- Most distribution ship the module as lua-lgi package.
-- libnm guide:   https://developer.gnome.org/libnm/1.0/
-- Lua-lgi guide: https://github.com/pavouk/lgi/blob/master/docs/guide.md
--

local lgi = require 'lgi'
local NM = lgi.NM

---------------------------
-- Main code starts here --
---------------------------
-- get client object
client = NM.Client.new()

print("Basic NM properties:")
print("====================")
print("NM version:          ", client.version)
print("NM state:            ", client.state)
print("NM startup:          ", client.startup)
print("Networking enabled:  ", client.networking_enabled)
print("Wireless enabled:    ", client.wireless_enabled)
print("Wireless HW enabled: ", client.wireless_hardware_enabled)
print("WWAN enabled:        ", client.wwan_enabled)
print("WWAN HW enabled:     ", client.wwan_hardware_enabled)
print("# devices:           ", #client.devices)
print("# connections:       ", #client.connections)
print("# active connections:", #client.active_connections)

File: examples/ruby/get-basic-nm-info.rb
#!/usr/bin/env ruby
# SPDX-License-Identifier: GPL-2.0-or-later
#
# Copyright (C) 2011 Red Hat, Inc.
#

require 'dbus'

#
# This example gets basic information about NetworkManager.
# Namely, it gets properties from /org/freedesktop/NetworkManager object.
#

# Get system bus
system_bus = DBus::SystemBus.instance

# Get the NetworkManager service
nm_service = system_bus.service("org.freedesktop.NetworkManager")

# Get the object from the service
nm_object = nm_service.object("/org/freedesktop/NetworkManager")

# Set default interface for the object
nm_object.default_iface = "org.freedesktop.NetworkManager"

# Introspect it
nm_object.introspect

properties = nm_object["org.freedesktop.DBus.Properties"].GetAll("org.freedesktop.NetworkManager")

puts "Basic NM properties:"
puts "===================="
properties[0].each do |prop,val|
  puts "#{prop} = #{val}"
end

